src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -913,9 +914,8 @@ void C2_MacroAssembler::string_compare(Register str1, Register str2,
       addi(cnt1, cnt1, 8);
     }
     addi(cnt2, cnt2, isUL ? 4 : 8);
+    bne(tmp1, tmp2, DIFFERENCE);
```

Comment:
You might want to place this `bne` before the preceding `addi(cnt2, cnt2, isUL ? 4 : 8);`

Reply:
Doubt so. Current code allows cnt2 to be calculated by the time the "bgez(cnt2, TAIL);" executed. So it benefits inorder dual-issue cpus (like hifive)